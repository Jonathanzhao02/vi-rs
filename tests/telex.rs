use std::fmt::Write;

fn snapshot_transform(input: &str) -> String {
    let mut result = String::new();
    for line in input.lines() {
        let words = line.trim().split_whitespace();

        let mut transformed_line = String::new();
        for word in words {
            let transformed_words = vi::telex::transform_buffer(word.chars());
            write!(&mut transformed_line, "{} ", transformed_words).unwrap();
        }
        write!(&mut result, "{}\n", transformed_line.trim()).unwrap();
    }
    result
}

mod shared;

macro_rules! gen_test_telex {
    ($name:tt, $path: tt) => {
        gen_test!(snapshot_transform, $name, $path);
    }
}
gen_test_telex!(simple_telex, "../testdata/input/simple_telex.txt");
