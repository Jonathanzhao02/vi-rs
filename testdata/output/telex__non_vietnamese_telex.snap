---
source: tests/telex.rs
expression: snapshot_transform(contents)
info:
  input_file: "../testdata/input/non_vietnamese_telex.txt"
---
samuwrite
c√°e
case
cat
doggo

